src/main/java/com/photolocos/enterprise/dao/PictureDAOStub.java

```diff
@@ -82,7 +85,16 @@ public void saveImage(MultipartFile image) throws IOException {
     }
 
     @Override
-    public PhotoDTO fetchByLocation(int locationID) {
-        return null;
+    public Set<PhotoDTO> fetchByLocation(LocationDTO location) {
+        List<PhotoDTO> photosCollection = new ArrayList<>(photos.values());
+        Set<PhotoDTO> matchedPhotos = new HashSet<>();
+
+        for (PhotoDTO photo : photosCollection) {
+            LocationDTO photoLocation = photo.getLocation();
+            if (photoLocation != null && photoLocation.equals(location)) {
+                matchedPhotos.add(photo);
+            }
+        }
+        return matchedPhotos;
```

Comment:
Implemented fetchByLocation method and changed parameter from locationId to a LocationDTO object. I also added an equals method to the LocationDTO class for comparing locations.